rotkehlchen/db/snapshots.py

```diff
@@ -0,0 +1,108 @@
+import csv
+import tempfile
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+from zipfile import ZipFile
+
+from rotkehlchen.accounting.export.csv import CSVWriteError
+from rotkehlchen.accounting.structures.balance import BalanceType
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.db.utils import DBAssetBalance, LocationData
+from rotkehlchen.types import Location, Timestamp
+
+
+class Snapshot:
+    def __init__(self, db_handler: DBHandler) -> None:
+        self.db_handler = db_handler
+        self.tmp_dir = tempfile.TemporaryDirectory()
+        self.tmp_dir_path = Path(self.tmp_dir.name)
+        self.snapshot_archive_path = self.tmp_dir_path / 'snapshot.zip'
+        self.timed_balances_csv_path = self.tmp_dir_path / 'timed_balances.csv'
+        self.timed_location_data_csv_path = self.tmp_dir_path / 'timed_location_data.csv'
+
+    def get_timed_balances(self, timestamp: Timestamp) -> List[Dict[str, Any]]:
+        balances_data = []
+        cursor = self.db_handler.conn.cursor()
+        timed_balances_result = cursor.execute(
+            'SELECT category, time, amount, currency, usd_value FROM timed_balances '
+            'WHERE time=?', (timestamp,),
+        )
+        for data in timed_balances_result:
+            balances_data.append(
+                DBAssetBalance(
+                    category=BalanceType.deserialize_from_db(data[0]),
+                    time=data[1],
+                    amount=data[2],
+                    asset=data[3],
+                    usd_value=data[4],
+                ),
+            )
+        return self.clean_data(balances_data)
+
+    def get_timed_location_data(self, timestamp: Timestamp) -> List[Dict[str, Any]]:
+        location_data = []
+        cursor = self.db_handler.conn.cursor()
+        timed_location_data = cursor.execute(
+            'SELECT time, location, usd_value FROM timed_location_data '
+            'WHERE time=?',
+            (timestamp,),
+        )
+        for data in timed_location_data:
+            location_data.append(
+                LocationData(
+                    time=data[0],
+                    location=data[1],
+                    usd_value=data[2],
+                ),
+            )
+
+        return self.clean_data(location_data)
+
+    def export(
+        self,
+        timed_balances_data: List[Dict[str, Any]],
+        timed_location_data: List[Dict[str, Any]],
+    ) -> Tuple[bool, str]:
```

Comment:
So this function is very wrong. You should have copied what we do with the accountant export or at least asked me. Look at what we do already and keep things consistent with the codebase. Please do:

1. Pass the timestamp to export
2. Pass an optional `directory_path: Optional[Path]`. If given then the CSV is exported to that directory. If not given a temporary directory is created.
3. Remove `timed_balances_data` and `timed_location_data` as arguments. These can be calculated inside the function itself.
4. Move the creation of the directory or usage of passed directory here.
5. Add a docstring explaining what this does and what it returns